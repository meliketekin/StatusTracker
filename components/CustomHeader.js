import React from "react";
import { View, StyleSheet, Text, TouchableOpacity } from "react-native";
import { Ionicons } from "@expo/vector-icons";

export const CustomHeader = (props) => {
  return (
    <View style={styles.header}>
      {props.hasGoBack && (
        <TouchableOpacity style={styles.backButton} onPress={props.onPress}>
          <Ionicons name="arrow-back" size={24} color="#2A2A2E" />
        </TouchableOpacity>
      )}
      <Text style={styles.title}>{props.text}</Text>
    </View>
  );
};

const styles = StyleSheet.create({
  header: {
    marginHorizontal: 20,
    height: 50,
    justifyContent: "center",
  },
  title: {
    fontFamily: "Poppins-SemiBold",
    textAlign: "center",
    color: "#2A2A2E",
    fontSize: 16,
  },
  backButton: {
    position: "absolute",
    padding: 10,
    margin: 5,
    borderWidth: 1,
    borderColor: "#E4E4E4",
    borderRadius: 15,
  },
});
