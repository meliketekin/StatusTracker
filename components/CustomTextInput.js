import React, { useRef, useEffect, useState } from "react";
import {
  View,
  TextInput,
  StyleSheet,
  Animated,
  Easing,
  TouchableOpacity,
} from "react-native";

export const CustomTextInput = (props) => {
  const transY = useRef(new Animated.Value(0));
  const transX = useRef(new Animated.Value(0));
  const scale = useRef(new Animated.Value(1));
  const [isFocused, setIsFocused] = useState(false);

  useEffect(() => {
    const isDropDown = props.isFocused && props.value.length > 0;
    if (isDropDown || isFocused) {
      handleFocus();
    }
  }, [props.isFocused, props.value, isFocused]);

  const handleFocus = () => {
    Animated.parallel([
      Animated.timing(transY.current, {
        toValue: -12,
        duration: 200,
        useNativeDriver: true,
        easing: Easing.ease,
      }),
      Animated.timing(scale.current, {
        toValue: 0.9,
        duration: 200,
        useNativeDriver: true,
        easing: Easing.ease,
      }),
      Animated.timing(transX.current, {
        toValue: -3,
        duration: 200,
        useNativeDriver: true,
        easing: Easing.ease,
      }),
    ]).start();
  };

  const handleBlur = () => {
    setIsFocused(false);
    if (props.value.length > 0) return;
    Animated.parallel([
      Animated.timing(transY.current, {
        toValue: 0,
        duration: 200,
        useNativeDriver: true,
        easing: Easing.ease,
      }),
      Animated.timing(scale.current, {
        toValue: 1,
        duration: 200,
        useNativeDriver: true,
        easing: Easing.ease,
      }),
      Animated.timing(transX.current, {
        toValue: 0,
        duration: 200,
        useNativeDriver: true,
        easing: Easing.ease,
      }),
    ]).start();
  };
  return (
    <View style={[styles.inputContainer, props.containerStyle]}>
      <Animated.View
        style={[
          styles.labelContainer,
          {
            transform: [
              { translateY: transY.current },
              { translateX: transX.current },
            ],
          },
        ]}
      >
        <Animated.Text
          style={[styles.label, { transform: [{ scale: scale.current }] }]}
        >
          {props.placeholder}
        </Animated.Text>
      </Animated.View>
      <TextInput
        value={props.value}
        style={styles.input}
        onFocus={() =>  setIsFocused(true)}
        onBlur={handleBlur}
        onChangeText={props.onChangeText}
        secureTextEntry={props.secureTextEntry}
        editable={props.isEditable}
      />
      {props.renderRightIcon && (
        <TouchableOpacity
          style={styles.eyeIcon}
          onPress={props.onPressRightIcon}
        >
          {props.renderRightIcon}
        </TouchableOpacity>
      )}
    </View>
  );
};

const styles = StyleSheet.create({
  inputContainer: {
    backgroundColor: "#FFFFFF",
    borderWidth: 1,
    borderColor: "#EAEBEC",
    borderRadius: 15,
    justifyContent: "center",
    paddingVertical: 5,
  },
  input: {
    fontFamily: "PlusJakartaSans-SemiBold",
    fontSize: 14,
    color: "#272A48",
    padding: 30,
    paddingLeft: 18,
    paddingTop: 25,
    paddingBottom: 10,
  },
  labelContainer: {
    position: "absolute",
    padding: 18,
  },
  label: {
    fontFamily: "PlusJakartaSans-Regular",
    fontSize: 14,
    color: "#28282866",
  },
  eyeIcon: {
    position: "absolute",
    padding: 18,
    right: 0,
  },
});
