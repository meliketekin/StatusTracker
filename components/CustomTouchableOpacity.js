import React from "react";
import { TouchableOpacity, StyleSheet, Text } from "react-native";
import { SCREEN_WIDTH } from "../constants/dimensions";
import { COLORS } from "../constants/colors";

export const CustomTouchableOpacity = (props) => {
  return (
    <TouchableOpacity
      style={[styles.button, props.containerStyle]}
      onPress={props.onPress}
    >
      <Text style={[styles.buttonText, props.textStyle]}>{props.text}</Text>
    </TouchableOpacity>
  );
};
const styles = StyleSheet.create({
  button: {
    width: SCREEN_WIDTH - 60,
    alignItems: "center",
    justifyContent: "center",
    paddingVertical: 20,
    borderRadius: 30,
    backgroundColor: COLORS.red,
  },
  buttonText: {
    fontFamily: "PlusJakartaSans-Bold",
    color: "#FFFFFF",
    fontSize: 15,
  },
});
