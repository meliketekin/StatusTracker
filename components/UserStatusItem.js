import { useNavigation } from "@react-navigation/native";
import React from "react";
import { StyleSheet, View, Text, Image, TouchableOpacity } from "react-native";

export const UserStatusItem = (props) => {
  const {user_status: status ,user_fullname: userName} = props.user
  const navigation = useNavigation();
  return (
    <TouchableOpacity style={styles.container} onPress={()=> navigation.navigate("EditUserScreen", {user: props.user})}>
      <View style={styles.userInfoContainer}>
        <Image
          source={require("../assets/images/user.png")}
          style={styles.userImage}
        />
        <Text style={styles.userName}>{userName}</Text>
      </View>
      <View style={[styles.statusContainer, {backgroundColor: status==="Active" ? "#D2FBD4": "#FFF0F0"}]}>
        <Text style={[styles.userStatus, {color: status==="Active" ? "#12AA18" : "#FF6464"}]}>{status}</Text>
      </View>
    </TouchableOpacity>
  );
};
const styles = StyleSheet.create({
  container: {
    backgroundColor: "#FFFFFF",
    flexDirection: "row",
    padding: 15,
    paddingVertical: 20,
    marginHorizontal: 20,
    borderRadius: 17,
    justifyContent: "space-between",
    alignItems: "center",
    marginBottom:10
  },
  userInfoContainer: {
    flexDirection: "row",
    alignItems: "center",
    gap: 15,
  },
  userImage: {
    width: 50,
    height: 50,
  },
  userName: {
    fontFamily: "PlusJakartaSans-SemiBold",
    fontSize: 14,
  },
  statusContainer: {
    padding: 8,
    paddingHorizontal:13,
    borderRadius: 30,
  },
  userStatus: {
    fontFamily: "PlusJakartaSans-Bold",
    fontSize:12
  },
});
