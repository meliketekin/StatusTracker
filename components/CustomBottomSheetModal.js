import { BottomSheetModal } from "@gorhom/bottom-sheet";
import { Text, View, Image, StyleSheet } from "react-native";
import React, { useRef, useMemo, useImperativeHandle } from "react";
import { BlurView } from "expo-blur";
import { Pressable } from "react-native";
import { CustomTouchableOpacity } from "./CustomTouchableOpacity";

export const CustomBottomSheetModal = React.forwardRef((props, ref) => {
  const snapPoints = useMemo(() => ["48%"], []);
  const modalRef = useRef(null);

  const renderBackdrop = ({ style }) => (
    <BlurView intensity={10} style={[style, { flex: 1 }]}>
      <Pressable onPress={modalRef.current?.close} style={{ flex: 1 }} />
    </BlurView>
  );

  useImperativeHandle(
    ref,
    function () {
      return {
        open() {
          modalRef.current?.present();
        },
      };
    },
    []
  );

  return (
    <BottomSheetModal
      ref={modalRef}
      index={0}
      snapPoints={snapPoints}
      backdropComponent={renderBackdrop}
    >
      <View
        style={{
          backgroundColor: "#FFFFFF",
          flex: 1,
          alignItems: "center",
          justifyContent: "center",
          gap:25
        }}
      >
        <View style={{ width: "30%", height: "30%" }}>
          <Image
            source={require("../assets/images/successImage.png")}
            resizeMode="contain"
            style={{ width: "100%", height: "100%" }}
          />
        </View>
        <Text style={styles.title}>Updated!</Text>
        <Text style={styles.subtitle}>Update Successfully.</Text>
        <CustomTouchableOpacity text="Okey"/>
      </View>
    </BottomSheetModal>
  );
});

const styles = StyleSheet.create({
  title: {
    fontFamily: "PlusJakartaSans-Bold",
    fontSize: 20,
    color: "#272A48",
  },
  subtitle: {
    fontFamily: "PlusJakartaSans-Medium",
    fontSize: 14,
    color: "#7F879E",
  },
});
