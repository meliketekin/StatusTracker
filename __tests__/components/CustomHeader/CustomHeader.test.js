import React from "react";
import { fireEvent, render } from "@testing-library/react-native";
import { CustomHeader } from "../../../components/CustomHeader";

describe("Custom Header tests", () => {
  it("should match with snapshot", () => {
    const header = render(<CustomHeader />);
    expect(header).toMatchSnapshot();
  });
  it("should render title successfully", () => {
    const text = "Header";
    const header = render(<CustomHeader text={text} />);
    const headerTitle = header.getByTestId("headerTitle").children[0];
    expect(headerTitle).toBe(text);
  });
  it("triggers onPress when the back button is pressed", () => {
    const onPress = jest.fn();
    const header = render(<CustomHeader hasGoBack={true} onPress={onPress} />);
    const headerBackButton = header.getByTestId("headerBackButton");
    fireEvent.press(headerBackButton);
    expect(onPress).toHaveBeenCalled();
  });
  it("should not display back button when hasGoBack prop is false", () => {
    const header = render(<CustomHeader hasGoBack={false} />);
    const headerBackButton = header.queryByTestId("headerBackButton");
    expect(headerBackButton).toBeNull();
  });
});
