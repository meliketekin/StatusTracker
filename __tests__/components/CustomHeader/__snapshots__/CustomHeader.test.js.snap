// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Custom Header tests should render successfully 1`] = `
<View
  style={
    {
      "height": 50,
      "justifyContent": "center",
      "marginHorizontal": 20,
      "marginTop": 30,
    }
  }
>
  <Text
    style={
      {
        "color": "#2A2A2E",
        "fontFamily": "Poppins-SemiBold",
        "fontSize": 16,
        "textAlign": "center",
      }
    }
    testID="headerTitle"
  />
</View>
`;
